### PR TITLE
Add segment to toolkit INTEG-1753

### PR DESCRIPTION
**Pull request by rusticpenguin**
## Purpose

Segment is tool we would like to use across all of our apps. This is the centralized solution where we can just install, add the provider, and you should be up and running!

Setup steps:
1. Import the provider
2. Add the props required
3. Call `trackEvent` from context, with the event and payload, wherever you'd like to track an event

### CODE REVIEW

**Comment by rusticpenguin** [comment]
Finishing up test right now. Will be added shortly

File: src/sdks/segment/segmentMarketplaceContext.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { SegmentEvent, SegmentEventData } from './types';
+import { createContext } from 'react';
+
+interface SegmentAnalyticsContextProps {
+  trackEvent: <T extends SegmentEvent>(action: T, trackingData?: SegmentEventData<T>) => void;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
```
Comment (matthew-gordon) [inline src/sdks/segment/segmentMarketplaceContext.tsx:9]: is this `noop` temporary, or is this leading to a pass through implementation eventually?

File: src/sdks/segment/segmentMarketplaceProvider.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      );
+    };
+
+    const { getByText } = render(
+      <SegmentAnalyticsProvider writeKey="testKey">
+        <TestChild />
+      </SegmentAnalyticsProvider>
+    );
+
+    await waitFor(() => expect(AnalyticsBrowser.load).toHaveBeenCalledTimes(1));
```
Comment (ethan-ozelius-contentful) [inline src/sdks/segment/segmentMarketplaceProvider.spec.tsx:59]: Hmmm.... I thought AnalyticsBrowser was being mocked?  I wouldn't have expected this assertion to succeed.  I would've expected `mockAnalytics.load` to have been called instead.

File: src/sdks/segment/segmentMarketplaceProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+interface SegmentAnalyticsProviderProps {
+  /**
+   * The Segment write key for your app. You can use `process.env.SEGMENT_WRITE_KEY`, `process.env.REACT_SEGMENT_WRITE_KEY`, `import.meta.env.VITE_SEGMENT_WRITE_KEY`, or manually set it here.
+   */
+  writeKey: string;
+  children: ReactNode;
+}
+
+/**
+ * A provider that sets up the Segment analytics SDK and provides a context for tracking events
```
Comment (ethan-ozelius-contentful) [inline src/sdks/segment/segmentMarketplaceProvider.tsx:20]: Not sure, but is this duplicate comment?  to comment is `interface SegmentAnalyticsProviderProps`

File: src/sdks/segment/utils/getUserCookieConsent.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { BaseAppSDK } from '@contentful/app-sdk';
+
+// TODO: remove these types once the SDK is updated with the new user consent types
+export declare const ACCEPT = 'ACCEPT';
```
Comment (ethan-ozelius-contentful) [inline src/sdks/segment/utils/getUserCookieConsent.ts:4]: what is declare doing here?

File: package-lock.json [hunk 5]
```diff
@@ -2478,10 +2481,9 @@
       }
     },
     "node_modules/@contentful/app-sdk": {
-      "version": "4.23.1",
-      "resolved": "https://npm.pkg.github.com/download/@contentful/app-sdk/4.23.1/2265db668071bba1cff09249e0bfcb83a4d7ca3c",
-      "integrity": "sha512-bDl3gz2Hw/7c+3yCkvs81/PO7c4WAv6xXGA6gCg2xWnkjrVvg5zMkGZxASzaqazevtikWg3fm5ktN5qfAT/iQQ==",
-      "license": "MIT",
+      "version": "4.24.0",
+      "resolved": "https://registry.npmjs.org/@contentful/app-sdk/-/app-sdk-4.24.0.tgz",
```
Comment (ethan-ozelius-contentful) [inline package-lock.json:2485]: shouldn't this point to github?  I remember running into something like this.  You might have to edit your `.npmrc` to point to the correct registry `@contentful:registry=https://npm.pkg.github.com`

**Review by ethan-ozelius-contentful — APPROVED** [review]
Left a couple non-blocking questions, the typewriter stuff is really interesting to me but I could definitely use some some context/hands-on experience.  Maybe a good walk through topic for code huddle or demo.

File: src/sdks/segment/segmentMarketplaceProvider.spec.tsx (hunk 2 above)
Comment (rusticpenguin) [reply src/sdks/segment/segmentMarketplaceProvider.spec.tsx]: Not sure how that worked lol

**Pull request merged**